app/models/binary/DataSetService.scala

```diff
@@ -271,22 +276,22 @@ class DataSetService @Inject()(organizationDAO: OrganizationDAO,
     }
 
   def publicWrites(dataSet: DataSet,
-                   userOpt: Option[User],
+                   requestingUserOpt: Option[User],
                    skipResolutions: Boolean = false,
-                   requestingUserTeamManagerMemberships: Option[List[TeamMembership]] = None): Fox[JsObject] = {
-    implicit val ctx = GlobalAccessContext
+                   requestingUserTeamManagerMemberships: Option[List[TeamMembership]] = None)(
+      implicit ctx: DBAccessContext): Fox[JsObject] =
```

Comment:
Quick comment to reviewer: Changing the AccessContext here to be more restrictive is not actually the fix (since the allowed teams db request has no AccessQuery), just something I noticed on the way.

Reply:
Probably it is not possible, but can't we just add an access query for the allowed teams request?

Reply:
I agree that that would be nice, however, the access query mechanism is not prepared for queries with joins in them. In those cases we filter/assert access in the scala code atm. Changing that would require changing _all_ the access queries, unfortunately